public/app/plugins/datasource/loki/types.ts

```diff
@@ -35,20 +35,13 @@ export enum LokiQueryDirection {
   Forward = 'forward',
 }
 
-export interface LokiQuery extends DataQuery {
-  queryType?: LokiQueryType;
-  expr: string;
+export interface LokiQuery extends Omit<LokiQueryFromSchema, 'queryType'> {
```

Comment:
Don't think `queryType` needs to be explicitly omitted here, since the code generator (when fixed for this case) will not do it and will extend `LokiQueryFromSchema` with overwriting `queryType`.

Reply:
hmm.. i understand that it is not necessary from the "schema" perspective, and also from the typescript perspective it works without it. but i prefer to be explicit here about the fact that we are redefining `queryType`, not just adding it.

or maybe i'm misunderstanding something here, will somehow the cue->typescript codegen literally not emit a `queryType:string` type for `DataQuery`?

Reply:
Currently there's a bug in cue where it uses the type from the parent DataQuery for query type, so will need to fix that and then there'll be no need to redefine queryType here.

Reply:
neat. done, i removed the `Omit` part.